internal/service/hub/handler/dsl/enforcer/score_maintainer.go

```diff
@@ -0,0 +1,154 @@
+package enforcer
+
+import (
+	"container/heap"
+	"sync"
+
+	"github.com/rss3-network/global-indexer/internal/service/hub/handler/dsl/model"
+	"github.com/samber/lo"
+)
+
+// priorityNodeQueue implements heap.Interface and holds NodeEndpointCaches.
+// It is used to maintain a priority queue of NodeEndpointCaches based on their scores and invalid counts.
+type priorityNodeQueue []*model.NodeEndpointCache
+
+func (pq priorityNodeQueue) Len() int {
```

Comment:
avoid using mixed receiver types (even though point receiver is not necessary here)

Reply:
refer to https://pkg.go.dev/container/heap#example-package-PriorityQueue

Reply:
> Don’t mix receiver types. Choose either pointers or struct types for all available methods. https://go.dev/wiki/CodeReviewComments

I believe this consistency avoids unnecessary errors